### ISSUE TITLE
Add a "stop" ?

### ISSUE THREAD

**Issue by excessnet** [bartbutenaers/node-red-contrib-interval-length#3]
I want almost exactly this node, execpt I want to calculate a "runtime" for something. What I would do is send the first message, then the second message and I want to know the time between those.

Later, if I receive another message, I want to start calculation until the next message and then output.

But right now, it will send an output as soon as the new message enter. 

What it does : 
[MSG1] -- interval -- [MSG2] -- interval -- [MSG3] -- interval -- [MSG4] 

What I want : 
[MSG1] -- interval -- [MSG2]
[MSG3] -- interval -- [MSG4]

Maybe if I could send "msg.reset true" it will stop counting and act like when it's deployed ?

Thanks !

**Comment by bartbutenaers** [comment]
Morning Excessnet,

Makes sense.  Perhaps I could add a 'reset' dropdown with 3 options:

- **None**: this is the default, so the user behaves like in the current NPM version.
- **Reset message**: in this case a `msg.reset` field in a reset message can reset the node.  The reset message is only used to reset the node, but it will be ignored during interval measurement (i.e. no interval is measured between the previous message and the reset message).
- **Reset in message**: in this case a `msg.reset` field in any message can reset the node.  The message will be used in the interval measurement like other messages (i.e. an interval is measured between the previous message and the message that contains a reset field).

Is that sufficient?  If so, give me a couple of days ...

P.S. could you explain a little bit more about how you are going to use this 'runtime' calculation.  Always nice if I can add some use cases to the readme file.

Kind regards,
Bart

**Comment by excessnet** [comment]
No rush, I made it in another way, but it would be a nice option. Even better with those options !

I would only use the option 2nd option "Reset message".

My thermostat is going "True" when heating and "False" when stop heating. I want to calculate the time between the true and the false so I know how long it has heated.

**Comment by bartbutenaers** [comment]
Hi Excessnet,

For simplicity, I have now added a checkbox to allow reset messages:

![image](https://user-images.githubusercontent.com/14224149/38167985-d31ce4ee-3540-11e8-9c89-f916962b43ea.png)

Here is an example flow:
![image](https://user-images.githubusercontent.com/14224149/38167951-331264b0-3540-11e8-98ed-97553280aeeb.png)
```
[{"id":"e6f4e8d1.34d3e8","type":"interval-length","z":"66fa353.41687cc","format":"mills","bytopic":false,"minimum":"","maximum":"","window":"","timeout":false,"msgTimeout":"","minimumunit":"msecs","maximumunit":"msecs","windowunit":"msecs","msgTimeoutUnit":"msecs","reset":true,"startup":false,"msgField":"payload","repeatTimeout":false,"name":"","x":1332,"y":180,"wires":[["60353310.88d95c"],[]]},{"id":"ef1fc202.21f8f","type":"inject","z":"66fa353.41687cc","name":"Normal message","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":940,"y":180,"wires":[["e6f4e8d1.34d3e8"]]},{"id":"60353310.88d95c","type":"debug","z":"66fa353.41687cc","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","x":1530,"y":174,"wires":[]},{"id":"5ea7c87a.c02488","type":"inject","z":"66fa353.41687cc","name":"Reset message","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":940,"y":220,"wires":[["dd66d070.e9ddc"]]},{"id":"dd66d070.e9ddc","type":"change","z":"66fa353.41687cc","name":"","rules":[{"t":"set","p":"reset","pt":"msg","to":"true","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":1140,"y":220,"wires":[["e6f4e8d1.34d3e8"]]}]
```

Could you please test this new option please?  You can install my test version directly from Github, by using following command:
**npm install bartbutenaers/node-red-contrib-interval-length**

When this fits your needs, I will publish a new version on NPM.  

Bart

**Comment by bartbutenaers** [comment]
This functionality is now available in version 0.0.3 on NPM.
